spark/src/main/java/com/nflabs/zeppelin/spark/SparkSqlInterpreter.java

```diff
@@ -76,13 +76,20 @@ public void open() {
     this.maxResult = conf.getInt("ZEPPELIN_SPARK_MAX_RESULT",
         "zeppelin.spark.maxResult",
         Integer.parseInt(getProperty("zeppelin.spark.maxResult")));
+    openSparkInterpreter(true);
```

Comment:
getSparkInterpreter returns SparkInterpreter object. Reopening spark Interpreter wont be safe. so this wont work:   

getSparkInterpreter().open();  
  
SparkInterpreter wrapped inside lazy interpreter. Calling open on lazy interpreter is safe because it opens the wrapper interpreter only if it has not been opened already. Here we are calling open on the Lazy Interpreter.